xls/examples/cocotb/running_counter.x

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import std
+
+proc RunningCounter {
+  base_r: chan<u32> in;
+  cnt_s: chan<u32> out;
+
+  init { u32:0 }
+
+  config(base_r: chan<u32> in, cnt_s: chan<u32> out) {
+    (base_r, cnt_s)
+  }
+
+  next(tok: token, cnt: u32) {
+    let (tok, base, base_valid) = recv_non_blocking(tok, base_r, u32:0);
+    let tok = send_if(tok, cnt_s, base_valid,  cnt + base);
```

Comment:
I think we want to send inconditionally even if we didn't receive a base (other wise the counter wouldn't run)

Reply:
I provided a simpler example to demonstrate cocotb usage